include/tc/core/execution_engine-inl.h

```diff
@@ -155,6 +155,42 @@ Duration ExecutionEngine<ExecutorType>::run(
   return res;
 }
 
+// Steal the executor instance and give it back under lock.
+// Run outside of lock on owning ExecutorType.
+template <typename ExecutorType>
+typename ExecutorType::ProfilingInfoType ExecutionEngine<ExecutorType>::profile(
```

Comment:
Let's please definitely **not** copy-paste this piece as it involves tricky concurrent code.
Let's instead factor out the code in another function and pass it a std::function since the only difference seems to be to call ```run``` vs ```profile```.